scripts/build/TestImpactAnalysis/storage_query_tool/local_storage_query_tool.py

```diff
@@ -106,16 +105,26 @@ def _put(self, file: str, storage_location: str):
         @param file: File to store.
         @param storage_location: Location to store the file.
         """
-        try:
-            json_obj = json.loads(file)
-            if not self._check_object_exists(str(storage_location)+"/historic_data.json"):
-                with open(f"{storage_location}/historic_data.json", "w", encoding="UTF-8") as raw_output_file:
-                    json.dump(json_obj, raw_output_file,
-                              ensure_ascii=False, indent=4)
-            else:
-                logger.info("Cancelling create, as file exists already")
-        except json.JSONDecodeError:
-            logger.error("The historic data does not contain valid json")
+        if self._file_type == self.FileType.JSON:
+            try:
+                json_obj = json.loads(file)
+                if not self._check_object_exists(str(storage_location)):
+                    with open(f"{storage_location}", "w", encoding="UTF-8") as raw_output_file:
+                        json.dump(json_obj, raw_output_file,
+                                ensure_ascii=False, indent=4)
+                else:
+                    logger.info("Cancelling create, as file exists already")
+            except json.JSONDecodeError:
+                logger.error("The historic data does not contain valid json")
+        if self._file_type == self.FileType.ZIP:
+            try:
+                if not self._check_object_exists(str(storage_location)):
```

Comment:
This logic is virtually identical to the logic of the JSON file type albeit the inner block of the `open`. Can these common code pathways be refactored into one?